layer/vector/KML.js

```diff
@@ -349,15 +327,50 @@ L.Util.extend(KML, {
 			}
 		}
 
-		if (name) {
-			layer.on('add', function(e) {
+		// NOTE: In newer versions of Leaflet we can listen to the add event for the
+		//       FeatureGroup also.
+		var layer = layers[0];
+		if (layers.length > 1) {
+			layer = new L.FeatureGroup(layers);
+			if (name) {
 				layer.bindPopup('<h2>' + name + '</h2>' + descr);
-			});
+			}
+		} else {
+			if (name) {
+				layer.on('add', function(e) {
+					layer.bindPopup('<h2>' + name + '</h2>' + descr);
+				});
+			}
 		}
 
 		return layer;
 	},
 
+	buildLayers: function (geomParent, xml, style, options) {
+		var layers = [];
+
+		// If we have any of the following return the layers for the contained geometry.
+		var multi = ['MultiGeometry', 'MultiTrack', 'gx:MultiTrack'];
+		for (h in multi) {
+		  el = geomParent.getElementsByTagName(multi[h]);
+		  for (i = 0; i < el.length; i++) {
+		    // Return geometry for first MultiGeometry that is found.
+		    return this.buildLayers(el[i], xml, style, options);
```

Comment:
Should we be adding to the list instead? We would need to return below if we find any of these geometry containers.